doc/user/compatibility.md

```diff
@@ -106,6 +106,14 @@ loop of your production application.
 
 ## C Extension Compatibility
 
+#### Identifiers may be macros or functions
+
+Identifiers which are normally macros may be functions, functions may be macros,
+and global variables may be macros. This causes problems where they are used in
+a context which relies on a particular implementation, for example a global
+variable assigned to an initial value which was a macro, like `Qnil`, which is
+a macro that evaluates to a function call in TruffleRuby.
```

Comment:
I think adding a specific example here (the `global = Qnil;` one) and a workaround would be very helpful here and clearer.

Reply:
#483